omnigibson/utils/bddl_utils.py

```diff
@@ -965,53 +1077,53 @@ def _import_sampleable_objects(self):
                            f"{valid_categories}"
 
-                for obj_inst in self._activity_conditions.parsed_objects[obj_synset]:
-                    # Don't explicitly sample if future
-                    if obj_inst in self._future_obj_instances:
-                        self._object_scope[obj_inst] = BDDLEntity(bddl_inst=obj_inst)
-                        continue
-                    # Don't sample if already in room
-                    if obj_inst in self._inroom_object_instances:
-                        continue
-
-                    # Shuffle categories and sample to find a valid model
-                    np.random.shuffle(categories)
-                    model_choices, category = set(), None
-                    for category in categories:
-                        # Get all available models that support all of its synset abilities
-                        model_choices = set(get_all_object_category_models_with_abilities(
-                            category=category,
-                            abilities=OBJECT_TAXONOMY.get_abilities(OBJECT_TAXONOMY.get_synset_from_category(category)),
-                        ))
-                        model_choices -= BAD_MODELS.get(category, set())
-                        if len(model_choices) > 0:
-                            break
-
-                    if len(model_choices) == 0:
-                        # We failed to find ANY valid model across ALL valid categories
-                        return f"Missing valid object models for all categories: {categories}"
-
-                    # Randomly select an object model
-                    model = np.random.choice(list(model_choices))
-
-                    # create the object
-                    simulator_obj = DatasetObject(
-                        name=f"{category}_{len(og.sim.scene.objects)}",
+                # Don't explicitly sample if future
+                if obj_inst in self._future_obj_instances:
+                    self._object_scope[obj_inst] = BDDLEntity(bddl_inst=obj_inst)
+                    continue
+                # Don't sample if already in room
+                if obj_inst in self._inroom_object_instances:
+                    continue
+
+                # Shuffle categories and sample to find a valid model
+                np.random.shuffle(categories)
+                model_choices = set()
+                for category in categories:
+                    # Get all available models that support all of its synset abilities
+                    model_choices = set(get_all_object_category_models_with_abilities(
                         category=category,
-                        model=model,
-                        prim_type=PrimType.CLOTH if "cloth" in OBJECT_TAXONOMY.get_abilities(obj_synset) else PrimType.RIGID,
-                    )
-                    num_new_obj += 1
+                        abilities=OBJECT_TAXONOMY.get_abilities(OBJECT_TAXONOMY.get_synset_from_category(category)),
```

Comment:
This code seems to change a lot from the previous iteration. Are we sure this doesn't break sampling?

Reply:
most of them are indentation changes. Yeah I tested some other task and I think it doesn't break.